File: AFSwinx/src/main/java/com/tomscz/afswinx/rest/rebuild/holder/AFDataHolder.java
package com.tomscz.afswinx.rest.rebuild.holder;

import java.util.HashMap;

/**
 * This class hold variables and their data representation. It hold class structure as well. This
 * object can be used to create original object include construct variable tree.
 * 
 * @author Martin Tomasek (martin@toms-cz.com)
 * 
 * @since 1.0.0.
 */
public class AFDataHolder {

    private String className;
    private HashMap<String, String> propertiesAndValues = new HashMap<String, String>();
    private HashMap<String, AFDataHolder> innerClasses = new HashMap<String, AFDataHolder>();

    public HashMap<String, AFDataHolder> getInnerClasses() {
        return innerClasses;
    }

    public HashMap<String, String> getPropertiesAndValues() {
        return propertiesAndValues;
    }

    public String getClassName() {
        return className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

    public void addInnerClass(AFDataHolder data) {
        innerClasses.put(data.getClassName(), data);
    }

    public void addPropertyAndValue(String propertyName, String value) {
        propertiesAndValues.put(propertyName, value);
    }

    public AFDataHolder getInnerClassByKey(String key) {
        return innerClasses.get(key);
    }

}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/rest/holder/AFDataHolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AFWindowsPhone.rest.holder
{
    class AFDataHolder
    {
        private String className;
        private Dictionary<String, String> propertiesAndValues = new Dictionary<String, String>();
        private Dictionary<String, AFDataHolder> innerClasses = new Dictionary<String, AFDataHolder>();

        public Dictionary<String, AFDataHolder> getInnerClasses()
        {
            return innerClasses;
        }

        public Dictionary<String, String> getPropertiesAndValues()
        {
            return propertiesAndValues;
        }

        public String getClassName()
        {
            return className;
        }

        public void setClassName(String className)
        {
            this.className = className;
        }

        public void addInnerClass(AFDataHolder data)
        {
            innerClasses.Add(data.getClassName(), data);
        }

        public void addPropertyAndValue(String propertyName, String value)
        {
            propertiesAndValues.Add(propertyName, value);
        }

        public AFDataHolder getInnerClassByKey(String key)
        {
            if (innerClasses.ContainsKey(key))
            {
                return innerClasses[key];
            }
            else
            {
                return null;
            }
        }
    }
}
